solr/core/src/java/org/apache/solr/util/circuitbreaker/MemoryCircuitBreaker.java

```diff
@@ -44,25 +48,20 @@ public boolean isCircuitBreakerGauntletTripped() {
       return false;
     }
 
-    allowedMemory = getCurrentMemoryThreshold();
+    allowedMemory.set(getCurrentMemoryThreshold());
 
-    seenMemory = calculateLiveMemoryUsage();
+    seenMemory.set(calculateLiveMemoryUsage());
 
-    return (seenMemory >= allowedMemory);
+    return (seenMemory.get() >= allowedMemory.get());
   }
 
   @Override
   public String printDebugInfo() {
-    return "seen memory=" + seenMemory + " allowed memory=" + allowedMemory;
+    return "seenMemory=" + seenMemory + " allowedMemory=" + allowedMemory;
```

Comment:
Need `.get()` here.

Reply:
Interesting, I thought the default toString delegate should handle the right debug output. Added the get, thanks